src/conn/conn_cache_pool.c

```diff
@@ -472,9 +472,9 @@ __cache_pool_assess(WT_SESSION_IMPL *session, uint64_t *phighest)
          * Figure out a delta since the last time we did an assessment for each metric we are
          * tracking. Watch out for wrapping of values.
          *
-         * Count pages read, assuming pages are 4KB.
+         * Count pages read, virtual memory page size.
          */
-        tmp = cache->bytes_read >> 12;
+        tmp = cache->bytes_read >> entry->page_size;
```

Comment:
Could you please elaborate on why we're using entry->page_size here?

Reply:
https://jira.mongodb.org/browse/WT-9965
We assume the page is always 4KB, which might not be the case. So I have changed it to be not hard coded. 
entry->page_size  accesses the virtual memory page size which I believe is what we want.

Reply:
Oopsies, fixed https://github.com/wiredtiger/wiredtiger/pull/10030/commits/5da302293cb33ab57a1fd82458caf88753446dd4

Reply:
It would be great to have a test for this change. I'll take a look and then we can discuss how that might be done.